thermalnetwork/ground_heat_exchanger.py

```diff
@@ -111,31 +97,84 @@ def get_common_inputs(self) -> dict:
     def get_input_file_data(self) -> dict:
         d = self.get_common_inputs()
 
-        # if (
-        #     self.design_method == GHEDesignType.AUTOSIZED_BIRECTANGLE_CONSTRAINED_BOREFIELD
-        #     or self.design_method == GHEDesignType.AUTOSIZED_BIZONED_RECTANGLE_BOREFIELD
-        #     or self.design_method == GHEDesignType.AUTOSIZED_NEAR_SQUARE_BOREFIELD
-        # ):
-        #     pass
-        if self.design_method == GHEDesignType.AUTOSIZED_RECTANGLE_BOREFIELD:
-            # if "polygons" in self.json_data["geometric_constraints"]:
-            #     geo_constraints = {
-            #         "property_boundary": self.json_data["geometric_constraints"]["polygons"][0],
-            #         "no_go_boundaries": self.json_data["geometric_constraints"]["polygons"][1:],
-            #         "b_min": self.json_data["geometric_constraints"]["b_min"],
-            #         "b_max_x": self.json_data["geometric_constraints"]["b_max"],
-            #         "b_max_y": self.json_data["geometric_constraints"]["b_max"],
-            #         "method": "BIRECTANGLECONSTRAINED",
-            #     }
-            # else:
-
-            geo_constraints = {
-                "length": self.json_data["borefield"]["length_of_ghe"],
-                "width": self.json_data["borefield"]["width_of_ghe"],
-                "b_min": self.json_data["borefield"]["b_min"],
-                "b_max": self.json_data["borefield"]["b_max"],
-                "method": "RECTANGLE",
-            }
+        if self.design_method in (
+            GHEDesignType.AUTOSIZED_NEAR_SQUARE_BOREFIELD,
+            GHEDesignType.AUTOSIZED_RECTANGLE_BOREFIELD,
+            GHEDesignType.AUTOSIZED_BIRECTANGLE_BOREFIELD,
+            GHEDesignType.AUTOSIZED_BIZONED_RECTANGLE_BOREFIELD,
+            GHEDesignType.AUTOSIZED_RECTANGLE_CONSTRAINED_BOREFIELD,
+            GHEDesignType.AUTOSIZED_BIRECTANGLE_CONSTRAINED_BOREFIELD,
+            GHEDesignType.AUTOSIZED_ROWWISE_BOREFIELD,
+        ):
+            if self.design_method == GHEDesignType.AUTOSIZED_NEAR_SQUARE_BOREFIELD:
+                geo_constraints = {
+                    "length": self.json_data["borefield"]["length_of_ghe"],
+                    "b": self.json_data["borefield"]["b"],
+                    "method": "NEARSQUARE",
+                }
+
+            elif self.design_method == GHEDesignType.AUTOSIZED_RECTANGLE_BOREFIELD:
+                geo_constraints = {
+                    "length": self.json_data["borefield"]["length_of_ghe"],
+                    "width": self.json_data["borefield"]["width_of_ghe"],
+                    "b_min": self.json_data["borefield"]["b_min"],
+                    "b_max": self.json_data["borefield"]["b_max"],
+                    "method": "RECTANGLE",
+                }
```

Comment:
Is there a need for the multi-level `if-else` statements here? Can we just flatten it out?

Reply:
We can flatten it but it will make the code longer and more redundant because this part is used by all of the `AUTOSIZED` options:

https://github.com/NREL/ThermalNetwork/pull/67/files#diff-cbade39a19de5e3726292f5293ba46b35c5125498e990140529b8642cea40c77R179-R196

I guess we can pull out that part into its own method that gets called at the end of each auutosize function. Would that be preferred?

Reply:
OK, don't worry about it then. We can deal with it later, if needed.